Test TestApplyMeetSinkError is unstable

### Which jobs are flaking?

unit_test
https://github.com/pingcap/ticdc/pull/2997

### Which test(s) are flaking?

TestApplyMeetSinkError


### Jenkins logs or GitHub Actions link

```log
[2021/10/28 21:38:24.448 +08:00] [INFO] [redo.go:100] ["apply redo log starts"] [checkpoint-ts=1000] [resolved-ts=2000]

[2021-10-28T13:41:04.417Z] [2021/10/28 21:38:24.455 +08:00] [INFO] [mysql.go:385] ["sink uri is configured"] ["format dsn"="root:******@tcp(127.0.0.1:4000)/?interpolateParams=true&multiStatements=true&allow_auto_random_explicit_insert=1&readTimeout=2m&tidb_txn_mode=optimistic&time_zone=%22UTC%22&timeout=2m&writeTimeout=2m"]

[2021-10-28T13:41:04.417Z] [2021/10/28 21:38:24.455 +08:00] [INFO] [mysql.go:588] ["Start mysql sink"]

[2021-10-28T13:41:04.417Z] [2021/10/28 21:38:25.459 +08:00] [INFO] [redo.go:100] ["apply redo log starts"] [checkpoint-ts=0] [resolved-ts=1]

[2021-10-28T13:41:04.417Z] --- FAIL: TestApplyMeetSinkError (0.01s)

[2021-10-28T13:41:04.417Z]     redo_test.go:239: 

[2021-10-28T13:41:04.417Z]         	Error Trace:	redo_test.go:239

[2021-10-28T13:41:04.417Z]         	Error:      	Expect "[CDC:ErrMySQLConnectionError]Open database connection failed: dial tcp 127.0.0.1:37390: connect: connection refused" to match "fail to open MySQL connection:.*connect: connection refused.*"

[2021-10-28T13:41:04.417Z]         	Test:       	TestApplyMeetSinkError

[2021-10-28T13:41:04.417Z] FAIL
```
https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_test/detail/cdc_ghpr_test/9947/pipeline


### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

--- a/pkg/applier/redo_test.go
+++ b/pkg/applier/redo_test.go
@@ -236,5 +236,5 @@ func TestApplyMeetSinkError(t *testing.T) {
 	}
 	ap := NewRedoApplier(cfg)
 	err = ap.Apply(ctx)
-	require.Regexp(t, "fail to open MySQL connection:.*connect: connection refused.*", err)
+	require.Regexp(t, "CDC:ErrMySQLConnectionError", err)
 }

